### ISSUE TITLE
Syncing and maintaining sync between 2 admins on raid

### ISSUE THREAD

**Issue by speaker** [kahluamods/konfersk#6]
This is a much more generalized problem perhaps, but we are having a difficult time keeping KSK on classic in sync. We're spending more time, uninstalling,  deleting SavedVariables, and then reloading then we are actually raiding. The problems are difficult to quantify other than to say that the loot master will run a successful bid, the item is delivered, the history is updated, the entry is moved briefly into the bottom of the list and it immediately pops back to the pre-bid position as if that position was pushed back to the loot master from another admin. The history is intact, but once this state happens, the list has to be completely flushed from everyone and rebuilt. We have set up only one list per configuration, (having more than one seemed to make the problem worse), and some configurations seems to behave fairly well, and others will simply fall apart after 1 bidding session.

My first ask is if there is a way to turn on some kind of debug logging that can be reviewed for potential problems. Honestly, I hope we're doing something wrong and the tool works because then we know we can use it once we learn proper use.

**Comment by speaker** [comment]
Pushing 9 months on this one, and over time this is gotten steadily worse. There are now only 2 admins listed, not even any alts, and every raid there's a 50-50 chance sync breaks and one whoever isn't the admin of a config has to uninstall and delete data and then be added back. Every guild I know on our server that was using KSK for looting is changing systems, not because they don't like Suicide Kings, but because they spend hours every week trying to get the admins back in sync. I'll submit some extra requests for specific asks.

**Comment by jkj** [comment]
Have you tried r17 that I just uploaded a few days ago? It isnt specifically for classic wow its meant for burning crusade but it will work on classic wow too you may have to enable "enable out of date mods". It has a new mechanism for compressing the data it sends on sync *AND* a mechanism for slowing data sends down - please see the changelog on curseforge. It is totally beyond my cntrol what is happening with syncs - this is code that has worked for 12 years that Blizzard fucked up when they put in checks to try and stop thread meters and LFG mods round about patch 1,13,2 or so. This new version does everything I was advised to do to try mittigate it. If you havent tried version 17 you really need to please - in my own tests I no longer get those "message was fake" or worse messages just being ignored stuff. Please let me know if you are still getting this with r17 and especially if you have tried "/ksk cps 400" and still get problems.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by speaker** [comment]
The problems we had never came with messages besides sometimes saying it was syncing. (Hence asking for debug.) It just stopped working in a number of possible states. Reloading UI sometimes was enough, other times one admin needed to uninstall and remove saved variables then setup again. (We tried every possible version of fixing, but deleting and re-installing was faster than other options.)

Honestly, we all thought the tool was abandoned. Just to get it to work pre-patch I updates enough code so it would at least run for the 2 week and I didn't even know you had changed it since it went "modified". 

If we continue with SK I'll certainly update.

**Comment by jkj** [comment]
Why would you think it was abandoned? I had a working version up on curseforge within 2 days of the pre-patch launching. Sorry if that wasn't quick enough for you :( Your original bug request that went unattended for 9 months was because you sent the bug report on github despite there being a pretty prominent message to send any bug reports to my email right there on the admin screen. I never look at github I just use it as a storage engine for git repos, which is exactly why I put in that message.

Just FYI I am going to release another version that will also compress normal syncing events, not just broadcasts, as those can be large too if you have an admin that is very out of date (in which case it really is safer to just remove them as an admin, broadcast, add them back and broadcast again).

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by speaker** [comment]
Well for one thing, I posted an issue on Sep 19, 2020 and it didn't get responded to till May 2021.
Because on the home page (http://kahluamod.com/ksk/) it says "Current version: r738 (released 2018-02-25)" right at the top.
I also don't see any prominent message about sending to your email so I'm not sure where I'm supposed to have found that. 

The basic raiding process has begun and we may use KSK again, if we do I can report back on what issues may have been resolved. In the past it was impossible to have an admin that was not in guild, (pug or guild mode) and if they were in guild it was almost always bad if they weren't on the raid with the other admins. During MC and BWL timeframe, the other loot master and I would spend about 2 hours fixing one or the other before raid time even though we were always on the same raid, and in the same guild. Over time the only fix was to export the ordered list, wipe all data, and reload the list after hand adding every user.

The single biggest frustration was the total lack of feedback from the system. We had no idea if it was still syncing or not, sometimes we would get a message, sometimes not. Having some kind of way to turn on messages so we would know that something was happening would have gone a long way in knowing if we just needed to be patient or if something went silently wrong.

I know this all sounds very negative, but the tool when it works is truly awesome and makes the LMs job much easier. The interface is intuitive, and simple and made many runs much smoother.

**Comment by jkj** [comment]
> Well for one thing, I posted an issue on Sep 19, 2020 and it didn't get responded to till May 2021.
Yup that's exactly the sort of thing that happens when you post to the wrong place. I do not use github for issue tracking. I use email. Nowhere in any documentation anywhere (or in the source) does it tell you to use github. You did that on your own.

> Because on the home page (http://kahluamod.com/ksk/) it says "Current version: r738 (released 2018-02-25)" right at the top.
Ummmm .... there is no WoW classic version there at all, the only place you have ever been able to get KSK for Classic is on CurseForge, which you must have done in order to be using the mod at all. What rational person would download something from site X and then go look on clearly outdated site Y for updates and then conclude things were abandoned based on that? Seriously man.

> I also don't see any prominent message about sending to your email so I'm not sure where I'm supposed to have found that.
/ksk config admin - could I possibly have made it any plainer to use email in that screen? Which you have to see regularly if you are messing with co-admins as much as you say you have?

> The single biggest frustration was the total lack of feedback from the system. We had no idea if it was still syncing or not, sometimes we would get a message, sometimes not. Having some kind of way to turn on messages so we would know that something was happening would have gone a long way in knowing if we just needed to be patient or if something went silently wrong.
It is 1000000 times more frustrating for me I can assure you. As you can no doubt tell I have put in a great deal of work into KSK to make it as intuitive and easy to use as possible. Do you really think, if I could do it, that I wouldn't give you some kind of progress report on a sync? The problem is that the WoW API provides ********* absolutely no mechanism for detecting how much of a message has been sent or if the other person has received it or not ********* So given that exactly how the hell am I supposed to give feedback? Sometimes the system itself is limited and there is nothing a mod author can do. If you are as experienced at mod authoring as previous messages claim you should know this.

Also please note that now it has been about 10 or 12 messages since I told you how I prefer to communicate (email, not github) and yet still you persist on using github. If you won't abide by a simple request to use the support platform of my choice please do not be surprised if your requests go unanswered for long periods of time or possibly even forever.

Also please note that the problems you listed with raiding were when there was a known problem with broadcasts being silently truncated. I tried to mitigate that using compression in the latest version but it doesn't sound like you are using that. For the record the latest version is r17-bcc. There is a small update coming soon when I release the EP/GP version of the mod. This really does seem to have solved most of the syncing problems but if you are not using it then you will not see any of the improvements.
‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐
